### PR TITLE
Fuzz the expr command

### PR DESCRIPTION
**Pull request by sylvestre**
Based on:
https://github.com/uutils/coreutils/pull/4641

### CODE REVIEW

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -149,6 +149,13 @@ jobs:
         ## Run it
         cd fuzz
         cargo +nightly fuzz run fuzz_test -- -max_total_time=${{ env.RUN_FOR }} -detect_leaks=0
+    - name: Run fuzz_expr for XX seconds
+      continue-on-error: true
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:153]: for now, it is failing but we don't fail the job on it. 
I opened :
https://github.com/uutils/coreutils/issues/5315
https://github.com/uutils/coreutils/issues/5316

but we probably have more issues

**Comment by sylvestre** [comment]
as expected, fails quickly:

```
 Different error codes
Expression: 60 + 60 | 9 < 38pkYJ * 45ÃhgÃ0 | 96 - 78 & 71
Rust code: 2
GNU code: 0
thread '<unnamed>' panicked at fuzz_targets/fuzz_expr.rs:167:17:
Different error codes
```

File: fuzz/fuzz_targets/fuzz_expr.rs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        } else {
+            let ch = invalid_utf8.choose(&mut rng).unwrap();
+            if let Some(c) = char::from_u32(*ch as u32) {
+                result.push(c);
+            }
+        }
+    }
+
+    result
+}
```
Comment (cakebaker) [inline fuzz/fuzz_targets/fuzz_expr.rs:65]: This function looks like a candidate for `fuzz_common.rs` as it is used in `fuzz_test.rs`, too.

File: fuzz/fuzz_targets/fuzz_expr.rs [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+    // Close the read end of the pipe
+    unsafe { libc::close(pipe_fds[0]) };
+
+    // Convert captured output to a string
+    let rust_output = String::from_utf8_lossy(&captured_output)
+        .to_string()
+        .trim()
+        .to_owned();
+
```
Comment (cakebaker) [inline fuzz/fuzz_targets/fuzz_expr.rs:157]: Lines 111 - 156 also look like a candidate for `fuzz_common.rs`, it seems to be identical to the code in `fuzz_test.rs`.

**Review by cakebaker — APPROVED** [review]
Looks good :) My suggestions are probably more something for a future PR.

**Comment by sylvestre** [comment]
agreed, it was part of my place to refactor some of the code :)

**Pull request merged**